lib/util.js

```diff
@@ -96,3 +96,8 @@ exports.nextTick = function(callback) {
     callback.apply(null, args);
   });
 };
+
+exports.clone = function(obj) {
+  return (obj === undefined) ? undefined : JSON.parse(JSON.stringify(obj));
+};
```

Comment:
This code has only hitherto been used in tests, and I wonder if we should use something more performant in production code?

If we're okay with adding a dependency, we've been using [`rfdc`](https://www.npmjs.com/package/rfdc) and it's worked well.

On the other hand, the performance gain is (relatively) small (~3x speed-up), so I don't mind just sticking with this.

@ericyhwang thoughts?

Reply:
>This code has only hitherto been used in tests

It was also used in `lib/db/memory.js`.

Reply:
The `MemoryDB` [shouldn't be used in production](https://github.com/share/sharedb/blob/3e636d0100921c26b4bb5108ef0afd1b27f838af/lib/db/memory.js#L7), so this clone method still hasn't touched production code before now.